File: labs/architecture-examples/angularjs_saltarelle/cs/services/TodoStorage.cs
using System.Collections.Generic;
using System.Html;
using System.Serialization;

namespace todo
{
    public class TodoStorage : ITodoStorage
    {
        private const string STORAGE_ID = "todos-angularjs-requirejs";

        public List<TodoItem> todos
        {
            get
            {
                var json = (string) Window.LocalStorage.GetItem(STORAGE_ID);
                return (List<TodoItem>)Json.Parse(json ?? "[]");
            }
            set
            {
                string json = Json.Stringify(todos);
                Window.LocalStorage.SetItem(STORAGE_ID, json);
            }
        }
    }
}

File: labs/architecture-examples/metajs-angular/mjs/services/todo_storage.mjs
(todomvc.factory 'todoStorage
                 (di-fn ()
                   (def storage-id "todos-angularjs")
                   {get: #(JSON.parse (or (localStorage.getItem storage-id) "[]"))
                    put: #(localStorage.setItem storage-id (JSON.stringify %))}))
